examples/edge-functions/supabase/functions/read-storage/index.ts

```diff
@@ -0,0 +1,64 @@
+// Follow this setup guide to integrate the Deno language server with your editor:
+// https://deno.land/manual/getting_started/setup_your_environment
+// This enables autocomplete, go to definition, etc.
+
+import { serve } from "https://deno.land/std@0.131.0/http/server.ts"
+import { createClient } from 'https://esm.sh/@supabase/supabase-js@2.0.0'
+
+const corsHeaders = {
+  'Access-Control-Allow-Origin': '*',
+  'Access-Control-Allow-Headers': 'authorization, x-client-info, apikey',
+}
+
+serve(async (req) => {
+  // read a text file from storage and print its contents
+  // This is needed if you're planning to invoke your function from a browser.
+  if (req.method === 'OPTIONS') {
+    return new Response('ok', { headers: corsHeaders })
+  }
+
+  try {
+    // Create a Supabase client with the Auth context of the logged in user.
+    const supabaseClient = createClient(
+      // Supabase API URL - env var exported by default.
+      Deno.env.get('SUPABASE_URL') ?? '',
+      // Supabase API ANON KEY - env var exported by default.
+      Deno.env.get('SUPABASE_ANON_KEY') ?? '',
+      // Create client with Auth context of the user that called the function.
+      // This way your row-level-security (RLS) policies are applied.
+      { global: { headers: { Authorization: req.headers.get('Authorization')! } } }
+    )
+
+    const { data, error } = await supabaseClient
+      .storage
+      .from('my-bucket')
+      .download('sample.txt')
+    if (error) throw error
+
+    // file contents are returned as a blob, we can convert it to utf-8 text by calling text() method.
+    const contents = await data.text();
+
+    // prints out the contents of the file
+    console.log(contents)
+
+    return new Response(JSON.stringify({ contents }), {
+      headers: { ...corsHeaders, 'Content-Type': 'application/json' },
```

Comment:
should we have another example for reading a generic blob like an image? and can we rename the current example as "reading a text file from storage". 

The content type here should be text/plain or is this example to read a JSON file?

Reply:
> should we have another example for reading a generic blob like an image? and can we rename the current example as "reading a text file from storage".

I think this example is good as it covers reading a file from Storage and then using that data in the function. Maybe a more advanced example would be reading a CSV.  Reading an image or other blob data isn't that useful as the user can't do much manipulation directly in the function.

> The content type here should be text/plain or is this example to read a JSON file?

The response content type can be JSON as we return a JSON object containing the file's `contents` as a field.

Reply:
The image example can still be useful for custom auth for storage. If you don't want to deal with RLS or you need to integrate with an external API for checking for permissions before returning an object from storage

Reply:
That makes sense! I'll leave that one to Storage team, as it'd need some custom policy examples on the Storage end as well.